src/mapblock.cpp

```diff
@@ -377,12 +416,20 @@ void MapBlock::serialize(std::ostream &os_compressed, u8 version, bool disk, int
 	const u8 params_width = 2;
 	if(disk)
 	{
-		MapNode *tmp_nodes = new MapNode[nodecount];
-		memcpy(tmp_nodes, data, nodecount * sizeof(MapNode));
-		getBlockNodeIdMapping(&nimap, tmp_nodes, m_gamedef->ndef());
+		MapNode *tmp_nodes;
+		if (m_is_mono_block) {
+			tmp_nodes = new MapNode[1];
+			tmp_nodes[0] = data[0];
+		}
+		else
```

Comment:
nit: this is generally not our style, this should be `} else {`

(and indeed this style is also used in other places throughout the file, so i wouldn't argue that there's a need to "blend in" here. rather i'd just fix the other occurrences.)

Reply:
:)

I do not feel strongly one way or the other. Both forms are readable to me.
Please decide which one to do and I'll make it so... :)